apps/dashboard/app/javascript/packs/files/file_ops.js

```diff
@@ -521,26 +521,33 @@ class FileOps {
     return Handlebars.compile(template_str);
   })();
 
-  poll(data) {
+  /**
+   * we pass myFileOp to the poller because we cannot 
+   * use "this" inside the poller to reference the object.
+   * If we instantiate a new instance of FileOp, we lose the ongoing count information.
+   */
+  poll(data, myFileOp) {
+
     $.getJSON(data.show_json_url, function (newdata) {
       // because of getJSON not being an actual piece of the object, we need to instantiate an instance FileOps for this section of code.
-      let myFileOp = new FileOps();
       myFileOp.findAndUpdateTransferStatus(newdata);
 
       if(newdata.completed) {
         if(! newdata.error_message) {
           if(newdata.target_dir == history.state.currentDirectory) {
             myFileOp.reloadTable();
           }
-
           // 3. fade out after 5 seconds
           myFileOp.fadeOutTransferStatus(newdata)
         }
-      }
-      else {
+      } else {
         // not completed yet, so poll again
         setTimeout(function(){
           myFileOp._attempts++;
```

Comment:
There's a bug here, I don't think we should ++ attempts here or rename the thing to failures.  In any case, you can poll all day long as long as the thing is going on it's all good. What we really care about is how many _failures_ have happened and we don't want to spam the system when it keeps failing.

Reply:
Another great observation!  I've implemented that from your sample code as well. Thanks again!!!

Reply:
Would you mind changing `_attempts` to `_failures` because that's what it really is?